src/objectExplorer/objectExplorerUtils.ts

```diff
@@ -40,8 +45,10 @@ export class ObjectExplorerUtils {
             return node.connectionCredentials.database;
         }
         while (node) {
-            if (node.nodeType === Constants.databaseString) {
-                return node.label;
+            if (node.metadata) {
```

Comment:
Hm, can you check if there's a bug here for DB-specific connections? The issue I'm seeing is if you're in a child node of the server tree that isn't under a database node you're going to get the wrong DB name.

- Server connection is to DB `mydb`
- Call getDatabaseName on a child node (such as Tables folder)
- It isn't a server node so goes into the while loop
- It isn't a database node and so walks up the tree, eventually getting to the server node
- We then return the default database label instead of the actual database for the connection

Seems like we need to still check if the server connection is to a specified db after falling out of the while loop

Reply:
@Charles-Gagnon That case wouldn't occur since we are looking at the metadata. Even though the rootNode is technically a server, the metadata will be a database. There wouldn't ever be a case when we are traversing a node's parent and we don't find a node's whose metadata isn't a `Database`

Reply:
Ah ok, well that's not intuitive at all! Thanks for looking into it though.